docs/adr/0004-support-hot-migration.md

```diff
@@ -0,0 +1,61 @@
+# 4. Support Hot Migration
+
+* Status: proposed
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1
+* Date: 2021-03-16
+
+Technical Story: \[description | ticket/issue URL\]
+
+## Context and Problem Statement
+
+Migration is one of the most important topics of the oCIS story. We need to provide a concept how to migrate from oC10 to oCIS.
+
+## Decision Drivers
+
+- Do not lose file blob or meta data.
+- Existing shares like public links and federated shares must remain functional after the migration.
+- Legacy clients need to keep working
+
+## Considered Options
+
+- Cold Migration
+- Hot Migration
+
+## Decision Outcome
+
+Chosen option: "\[option 1\]", because \[justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)\].
+
+### Positive Consequences
+
+- \[e.g., improvement of quality attribute satisfaction, follow-up decisions required, …\]
+- …
+
+### Negative Consequences
+
+- \[e.g., compromising quality attribute, follow-up decisions required, …\]
+- …
+
+## Pros and Cons of the Options
+
+### Cold Migration
+
+The migration happens while the service is offline. File metadata, blobs and share data is exported from ownCloud 10 and imported in oCIS.
```

Comment:
Is data exported completely, ie. for all users, storages, incl. Trash and Versions etc? Or can that be done partially?